### PR TITLE
Bed 4538

### PR DESCRIPTION
**Pull request by stephanieslamb**
## Description

Updated the existing GET ListSavedQueries endpoint to return public, shared and owned queries.  Added a query param, scope, which can return only public, shared or owned queries as well as any combination of scope types.  

## Motivation and Context

This PR addresses: BED-4538

*Why is this change required? What problem does it solve?*
This ticket is part of the on going initiative to improve saved queries.  

## How Has This Been Tested?

This has been tested via curl commands(see screenshots below) as well as unit tested. 

## Screenshots (optional):

Owned saved queries
Screenshot 2024-08-06 at 4 59 26 PM

Shared and public queries
Screenshot 2024-08-06 at 4 59 39 PM

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist:

- [X] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [X] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [X] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: cmd/api/src/api/v2/saved_queries.go
Comment (superlinkx) [inline cmd/api/src/api/v2/saved_queries.go:123]: Add return after to ensure we are closing the connection after error

File: cmd/api/src/api/v2/saved_queries.go
Comment (superlinkx) [inline cmd/api/src/api/v2/saved_queries.go:88]: Move logic to ParseQueryParamFilter function as a carveout, like we do with pagination. This keeps our special logic in one place even if it's similarly hacky.

File: cmd/api/src/api/v2/saved_queries_test.go
Comment (superlinkx) [inline cmd/api/src/api/v2/saved_queries_test.go:287]: If we change these to:

```go
req, err := http.NewRequestWithContext(createContextWithOwnerId(userId), "GET", endpoint, nil)
require.Nil(t, err)

q := url.Values{}
...
```

We can avoid indenting logic flows and better see what points are required to run the test

File: cmd/api/src/api/v2/saved_queries_test.go
Comment (superlinkx) [inline cmd/api/src/api/v2/saved_queries_test.go:305]: I always recommend using assert for your actual test criteria, as they will not fatally stop the test. In this case, the test has no logic after a single assert, but in the future there may be additional test cases or possible cleanup logic and changing these to asserts will help out the future engineer working on the test.

**Review by superlinkx — CHANGES_REQUESTED** [review]

**Comment by superlinkx** [comment]
Test changes aren't required, they're complete nits. The other two include a nice to have and a requirement to ensure we don't try to send more than one API response back. Once that's fixed, I'm happy to approve these changes as they're otherwise perfect!

File: cmd/api/src/api/v2/saved_queries_test.go
Comment (stephanieslamb) [reply cmd/api/src/api/v2/saved_queries_test.go:287]: updated!

**Review by superlinkx — APPROVED** [review]

File: cmd/api/src/api/v2/saved_queries.go
Comment (mistahj67) [inline cmd/api/src/api/v2/saved_queries.go:113]: If these are the same scopes as https://github.com/SpecterOps/BloodHound/blob/3a68a2240e716d4fd9a89fad77c014d7d03cee78/cmd/api/src/database/saved_queries_permissions.go

We should use the constants as those are Title case and these are lowercase

Comment (stephanieslamb) [reply cmd/api/src/api/v2/saved_queries.go:113]: They are the same! I didn't know those had been added in Ari's PR.  Since her PR isn't merged yet, would you like me to go ahead and add them in mine? Or wait until her PR has been merged?

Comment (superlinkx) [reply cmd/api/src/api/v2/saved_queries.go:113]: You may want to create separate constants in this file, as they are the same, but different (title vs lowercase, and different layers). I guess we could use the same constants if we were to `toLower` the constant before checking it against scope (may want to `toLower` the scope var in the switch too)

Comment (superlinkx) [reply cmd/api/src/api/v2/saved_queries.go:113]: Actually, reviewing those constants, I'm not sure they should be title case. We could probably get those changed to lowercase and reuse them. It wouldn't hurt to make sure we're setting the incoming `scope` parameter to lowercase before checking against our strings/constants either way though

Comment (stephanieslamb) [reply cmd/api/src/api/v2/saved_queries.go]: Updated this.  Let me know what you both think.  I did just add the consts that are in Ari's PR.

Comment (superlinkx) [reply cmd/api/src/api/v2/saved_queries.go]: This is great. Ship it.

**Review by superlinkx — APPROVED** [review]

**Pull request merged**
